server/src/test/java/io/spine/examples/shareaware/server/wallet/WalletWithdrawalTest.java

```diff
@@ -0,0 +1,404 @@
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.shareaware.server.wallet;
+
+import io.spine.examples.shareaware.WalletId;
+import io.spine.examples.shareaware.WithdrawalId;
+import io.spine.examples.shareaware.paymentgateway.command.TransferMoneyToUser;
+import io.spine.examples.shareaware.paymentgateway.event.MoneyTransferredToUser;
+import io.spine.examples.shareaware.server.given.WithdrawalTestContext;
+import io.spine.examples.shareaware.server.given.ControllablePaymentGatewayProcess;
+import io.spine.examples.shareaware.server.paymentgateway.PaymentGatewayProcess;
+import io.spine.examples.shareaware.wallet.Wallet;
+import io.spine.examples.shareaware.wallet.WalletBalance;
+import io.spine.examples.shareaware.wallet.WalletWithdrawal;
+import io.spine.examples.shareaware.wallet.command.CancelMoneyReservation;
+import io.spine.examples.shareaware.wallet.command.DebitReservedMoney;
+import io.spine.examples.shareaware.wallet.command.ReserveMoney;
+import io.spine.examples.shareaware.wallet.command.WithdrawMoney;
+import io.spine.examples.shareaware.wallet.event.MoneyNotWithdrawn;
+import io.spine.examples.shareaware.wallet.event.MoneyReservationCanceled;
+import io.spine.examples.shareaware.wallet.event.MoneyReserved;
+import io.spine.examples.shareaware.wallet.event.MoneyWithdrawn;
+import io.spine.examples.shareaware.wallet.event.ReservedMoneyDebited;
+import io.spine.examples.shareaware.wallet.rejection.Rejections.InsufficientFunds;
+import io.spine.money.Money;
+import io.spine.server.BoundedContextBuilder;
+import io.spine.testing.server.blackbox.ContextAwareTest;
+import io.spine.testing.server.model.ModelTests;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static io.spine.examples.shareaware.server.given.WalletTestEnv.*;
+import static io.spine.examples.shareaware.server.wallet.MoneyCalculator.*;
+
+@DisplayName("`WalletWithdrawal` should")
+public final class WalletWithdrawalTest extends ContextAwareTest {
+
+    @BeforeAll
+    static void beforeAll() {
+        ModelTests.dropAllModels();
+    }
+
+    @Override
+    protected BoundedContextBuilder contextBuilder() {
+        return WithdrawalTestContext.newBuilder();
+    }
+
+    @Nested
+    @DisplayName("reduce the wallet balance")
+    class WithdrawMoneyFromBalance {
+
+        @Test
+        @DisplayName("for 400 USD")
```

Comment:
How do we know that's $400? It's nowhere near in the code of this test.

In such cases, please don't refer to exact test values in display names. It's difficult to verify (and maintain later).